### PR TITLE
Duplicate checker only checks for edited fields

### LINKED ISSUE #95: Handle duplicates when editing a contact field
Summary:
Duplicate handling in `editContactCommand` checks all the fields for similarities instead of just checking fields that are being edited. As a result, as long as any two contacts have similar fields, any edits done to them will always warn user of duplicates.

For example, if one Contact has name "Bory" with email "bory@gmail.com" and another Contact has name "Marcus" with email  "bory@gmail.com", editing any of their names will raise a duplicate warning because of their email.

### PR DESCRIPTION
**Pull request by ashrafjfr**
Fixes #95 

New additions to this PR:
- Uses a Boolean array to check which fields are edited
- Only fields with edits will be checked for duplicates

### CODE REVIEW

File: src/main/java/seedu/command/EditContactCommand.java [hunk 1]
```diff
@@ -67,6 +72,14 @@ private Boolean hasDuplicates(Contact postEditContact, ContactList contactList,
         return false;
     }
 
+    private Boolean[] hasEditedFieldChecker(String[] contactDetails) {
```
Comment (marcusbory) [inline src/main/java/seedu/command/EditContactCommand.java:75]: Minor edit, apologies for being anal:
```suggestion
    private boolean[] hasEditedFields(String[] contactDetails) {
```

File: src/main/java/seedu/command/EditContactCommand.java [hunk 2]
```diff
@@ -43,21 +45,24 @@ public void execute() {
     private Boolean hasDuplicates(Contact postEditContact, ContactList contactList, int contactIndex)
             throws InvalidFlagException {
         ArrayList<Integer> duplicatedIndex = new ArrayList<>();
+        Boolean[] hasEditedField = hasEditedFieldChecker(contactDetails);
```
Comment (marcusbory) [inline src/main/java/seedu/command/EditContactCommand.java:48]: Hi @ashrafjfr , I believe you should be able to use `boolean` as a primitive Java data type. Capitalised `Boolean` is an object (similar to `int` vs `Integer`). Generally, in the case of using `Boolean`, class methods are used. However, in your implementation, I think that `boolean` primitive data type should suffice. In addition, a primitive data type uses a lot less memory than an object.

tl;dr - In your use case, use `boolean` instead of `Boolean`.
```suggestion
        boolean[] hasEditedFields = hasEditedFieldChecker(contactDetails);
```

File: src/main/java/seedu/command/EditContactCommand.java [hunk 3]
```diff
@@ -67,6 +72,14 @@ private Boolean hasDuplicates(Contact postEditContact, ContactList contactList,
         return false;
     }
 
+    private Boolean[] hasEditedFieldChecker(String[] contactDetails) {
+        Boolean[] hasEditedField = new Boolean[NUMBER_OF_FIELDS];
+        for (int i = 0; i < NUMBER_OF_FIELDS; i++) {
+            hasEditedField[i] = (contactDetails[i] != null);
+        }
+        return hasEditedField;
+    }
```
Comment (marcusbory) [inline src/main/java/seedu/command/EditContactCommand.java:81]: Good, simple implementation!
